vividus-plugin-azure-service-bus/src/main/java/org/vividus/azure/servicebus/ServiceBusSteps.java

```diff
@@ -76,6 +80,47 @@ public void sendMessageToServiceBus(String serviceBusKey, String payload)
         serviceBusService.send(serviceBusKey, payload);
     }
 
+    /**
+     * Send message with custom properties to
+     * <a href="https://learn.microsoft.com/en-us/azure/service-bus-messaging">Azure Service Bus</a>
+     * <br>
+     * <br>Service Bus <b>connection configuration</b> should be defined via properties with a following format:
+     * <br><code>azure.service-bus.{service-bus-key}.{property-name}</code>
+     * <br>Example:
+     * <br><code>
+     * <br>azure.service-bus.test-topic.channel-type=TOPIC
+     * <br>azure.service-bus.test-topic.namespace=testNamespace
+     * <br>azure.service-bus.test-topic.name=testTopicName
+     * <br>azure.service-bus.test-topic.subscription-name=testSubscriptionName
+     * </code>
+     * <br>
+     * <br> Where:
+     * <ul>
+     *     <li><b>channel-type</b> - the type of service bus messaging components: either `QUEUE` or `TOPIC`</li>
+     *     <li><b>namespace</b> - the name of the namespace the service bus belongs to</li>
+     *     <li><b>name</b> - the queue or topic name</li>
+     *     <li><b>subscription-name</b> - the name of the topic subscription. Only for TOPIC channel-type</li>
+     * </ul>
+     * <p>The custom properties to be defined in the ExamplesTable</p>
+     * <ul>
+     *      <li><b>key</b> - the property name</li>
+     *      <li><b>type</b> - the data type of the property ('String', 'Boolean', 'Number')</li>
+     *      <li><b>value</b> - the property value</li>
+     * </ul>
+     * @param serviceBusKey The key associated with Azure Service Bus connection configuration.
+     * @param payload       Message to send to the service bus.
+     * @param customProperties The table of custom properties to include with the message.
+     */
+    @When("I send message to `$serviceBusKey` service bus with payload:`$payload` and custom properties:"
+            + "$customProperties")
+    public void sendMessageToServiceBus(String serviceBusKey, String payload, ExamplesTable customProperties)
+    {
+        Map<String, Object> properties = new HashMap<>();
+        customProperties.getRows().forEach(
+                row -> properties.put(row.get("key"), castPropertyType(row.get("type"), row.get("value"))));
```

Comment:
use Java stream converting rows to Map directly (using `toMap` without intermediate creation of HashMap)